src/main/java/com/fasterxml/jackson/databind/jsontype/impl/TypeNameIdResolver.java

```diff
@@ -80,7 +82,16 @@ public static TypeNameIdResolver construct(MapperConfig<?> config, JavaType base
     @Override
     public String idFromValue(Object value)
     {
-        Class<?> cls = _typeFactory.constructType(value.getClass()).getRawClass();
+       return idFromClass(value.getClass());
+    }
+
+
+    protected String idFromClass(Class<?> clazz)
+    {
+        if(clazz==null){
```

Comment:
Correct, I am just trying to understand what is the intended change -- not saying it breaks anything that wasn't already broken. But I guess the answer is seen below... so never mind.

I am ok without changes.